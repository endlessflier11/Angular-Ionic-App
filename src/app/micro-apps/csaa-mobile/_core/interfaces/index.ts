export * from './auth.interface';
export * from './router';
export * from './http.interface';
export * from './vehicle.interface';
export * from './ionic-lifecycle.interface';
export * from './analytics.interface';
export * from './policy.interface';
export * from './payment.interface';
export * from './config.interface';
export * from './ngxs-store.interface';
