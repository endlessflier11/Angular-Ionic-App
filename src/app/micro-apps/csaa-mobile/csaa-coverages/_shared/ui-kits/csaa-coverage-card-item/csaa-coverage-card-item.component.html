<div
  (click)="handleItemClick($event)"
  class="csaa-coverage-card-item"
  [class.open]="isOpen && coverage.shortDesc"
>
  <h4 class="title">{{ coverage.shortName }}</h4>
  <div *ngIf="!isOpen || !coverage.shortDesc" class="unopened-card">
    <ng-container
      *ngIf="coverage.coverages && coverage.coverages.length > 0; else noIndentedCoverage"
    >
      <ion-item
        detail
        track-click="coverages:gotoindentedcoverages"
        (click)="goToIndentedCoverages($event)"
      >
        <h3>Tap to view coverages</h3>
      </ion-item>
    </ng-container>
    <ng-template #noIndentedCoverage>
      <p class="limit-info">
        <ng-container *ngIf="coverage.general; else otherCoverage">
          <ng-container *ngIf="coverage.individualLimit; else noIndividualLimit">
            <span>{{ coverage.individualLimit | currency: 'USD':'symbol':'1.0-0' }}</span>
            {{ coverage.individualLimitDelimiter }}
            <ng-container *ngIf="coverage.occurrenceLimit">
              <span>/ {{ coverage.occurrenceLimit | currency: 'USD':'symbol':'1.0-0' }}</span>
              {{ coverage.occurrenceLimitDelimiter }}
            </ng-container>
          </ng-container>
          <ng-template #noIndividualLimit>
            <ng-container *ngIf="coverage.occurrenceLimit">
              <span>{{ coverage.occurrenceLimit | currency: 'USD':'symbol':'1.0-0' }}</span>
              {{ coverage.occurrenceLimitDelimiter }}
            </ng-container>
          </ng-template>
        </ng-container>

        <ng-template #otherCoverage>
          <ng-container *ngIf="policyType === PolicyType.PUP; else vehicleCoverage">
            <span>{{ coverage.combinedSingleLimitAmount | currency: 'USD':'symbol':'1.0-0' }}</span>
          </ng-container>
          <ng-template #vehicleCoverage>
            <ng-container *ngIf="coverage.covered">
              <ng-container *ngIf="isComprehensiveCoverage()">
                <span>{{ coverage.occurrenceLimit | currency: 'USD':'symbol':'1.0-0' }}</span>
                {{ coverage.occurrenceLimitDelimiter }}
                <ng-container *ngIf="isGlassIndicatorSet()">
                  <span>/ $0</span> Glass Deductible
                </ng-container>
              </ng-container>
              <ng-container *ngIf="!isComprehensiveCoverage()">
                <ng-container *ngIf="!coverage.individualLimit && coverage.occurrenceLimit">
                  <span>{{ coverage.occurrenceLimit | currency: 'USD':'symbol':'1.0-0' }}</span>
                  {{ coverage.occurrenceLimitDelimiter }}
                </ng-container>
                <ng-container *ngIf="coverage.individualLimit && !coverage.occurrenceLimit">
                  <span>{{ coverage.individualLimit | currency: 'USD':'symbol':'1.0-0' }}</span>
                  {{ coverage.individualLimitDelimiter }}
                </ng-container>
                <ng-container *ngIf="coverage.individualLimit && coverage.occurrenceLimit">
                  <span
                    >{{ coverage.individualLimit | currency: 'USD':'symbol':'1.0-0' }}/{{
                      coverage.occurrenceLimit | currency: 'USD':'symbol':'1.0-0'
                    }}</span
                  >
                  {{ coverage.individualLimitDelimiter }}/{{ coverage.occurrenceLimitDelimiter }}
                </ng-container>
                <ng-container
                  *ngIf="
                    policyType === PolicyType.Auto &&
                    coverage.deductible &&
                    coverage.deductible.isPresent
                  "
                >
                  <span>{{ coverage.deductible.value | currency: 'USD':'symbol':'1.0-0' }}</span>
                  Deductible
                </ng-container>
              </ng-container>
            </ng-container>
            <ng-container *ngIf="!coverage.covered">
              {{ coverage.twitterPitch }}
            </ng-container>
          </ng-template>
        </ng-template>
      </p>
    </ng-template>
  </div>
  <div *ngIf="isOpen && coverage.shortDesc" class="description">{{ coverage.shortDesc }}</div>
</div>
