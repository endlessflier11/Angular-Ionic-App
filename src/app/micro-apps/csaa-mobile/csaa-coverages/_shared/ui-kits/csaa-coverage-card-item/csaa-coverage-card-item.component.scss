@import '../../../../_scss/variables';
@import '../../../../_scss/mixins';

.title {
  color: var(--ion-color-primary, $csaa-active-color);
  font-weight: $csaa-bolder-font-weight;
  font-size: 16px;
  margin: 10px 0;
}

p.limit-info {
  font-size: 12px;
  margin-bottom: 12px;
  font-weight: $csaa-light-font-weight;
  span {
    font-weight: $csaa-active-font-weight;
  }
}

.open {
  // width: 110%;
  // transform: translateX(-4.55%);
  z-index: 10;
  padding-top: 1px;
  padding-left: 5px;

  // @include border-radius($csaa-card-border-radius);
  @include box-shadow(10px 0px 20px 0px rgba(0, 0, 0, 0.2));

  background: -moz-linear-gradient(
    top,
    $csaa-card-gradient-start 0%,
    $csaa-card-gradient-end 100%
  ); /* FF3.6-15 */
  background: -webkit-linear-gradient(
    top,
    $csaa-card-gradient-start 0%,
    $csaa-card-gradient-end 100%
  ); /* Chrome10-25,Safari5.1-6 */
  background: linear-gradient(
    to bottom,
    $csaa-card-gradient-start 0%,
    $csaa-card-gradient-end 100%
  ); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
}

.description {
  padding: 4px 16px 16px 25px;
  font-size: 12px;
}

// .csaa {
//   csaa-coverage-card-item,
//   csaa-indented-coverage-card-item {
//     h2 {
//       color: $csaa-active-color;
//       font-weight: $csaa-active-font-weight;
//     }
//     .twitter-pitch {
//       font-family: $csaa-subtitle-font-family-ios;
//       font-weight: $csaa-light-font-weight;
//     }
//     .home-limit {
//       margin-top: 0.3rem;
//       font-family: $csaa-subtitle-font-family-ios;
//       font-weight: $csaa-light-font-weight;
//       span {
//         font-weight: $csaa-bold-font-weight;
//       }
//     }
//     &.no-border {
//       ion-item {
//         border: none;
//       }
//     }
//   }
// }
