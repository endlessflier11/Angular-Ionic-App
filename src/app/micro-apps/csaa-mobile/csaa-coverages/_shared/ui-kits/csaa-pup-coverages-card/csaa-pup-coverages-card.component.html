<csaa-skeleton-card *ngIf="loading"></csaa-skeleton-card>
<ion-card *ngIf="!loading && coverages && coverages.length" class="coverage-list">
  <ion-card-header>
    <ion-card-title>
      <div class="display-flex ion-align-items-center">
        {{ policyNumber }}
      </div>
      <csaa-theme-icon [type]="policy?.type" [size]="40"> </csaa-theme-icon>
    </ion-card-title>
    <ion-card-subtitle>Personal Umbrella Policy</ion-card-subtitle>
  </ion-card-header>

  <ion-card-content class="csaa-item-list">
    <ion-list *ngIf="coverages.length > 0">
      <ion-item *ngFor="let coverage of coverages; let ix = index">
        <csaa-coverage-card-item
          [policyType]="PolicyType.PUP"
          *ngFor="let coverage of coverages; let ix = index"
          [coverage]="coverage"
          (itemClick)="onItemClick(coverage, ix)"
          [isOpen]="isItemOpen(ix)"
        >
        </csaa-coverage-card-item>
      </ion-item>
    </ion-list>
    <ion-button expand="full" (click)="viewDeclarationClick.emit($event)">
      View Declarations</ion-button
    >
  </ion-card-content>
</ion-card>
