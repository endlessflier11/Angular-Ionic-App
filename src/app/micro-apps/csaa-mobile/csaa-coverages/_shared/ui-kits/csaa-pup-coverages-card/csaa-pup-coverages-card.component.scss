ion-card-title {
  display: flex;
  flex-flow: row wrap;
  div {
    flex-grow: 1;
  }
}
