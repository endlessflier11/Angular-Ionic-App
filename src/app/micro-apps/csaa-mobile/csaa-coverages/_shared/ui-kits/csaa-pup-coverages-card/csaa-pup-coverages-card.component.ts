import { Component, EventEmitter, Input, Output } from '@angular/core';
import { Category, EventName } from '../../../../_core/interfaces/analytics.interface';
import { PolicyType } from '../../../../_core/interfaces/policy.interface';
import { AnalyticsService } from '../../../../_core/services/analytics.service';
import { CsaaCoverageListCardBaseComponent } from '../csaa-coverage-list-card-base/csaa-coverage-list-card-base.component';

@Component({
  selector: 'csaa-pup-coverages-card',
  templateUrl: './csaa-pup-coverages-card.component.html',
  styleUrls: ['./csaa-pup-coverages-card.component.scss'],
})
export class CsaaPupCoveragesCardComponent extends CsaaCoverageListCardBaseComponent {
  @Output() viewDeclarationClick: EventEmitter<any> = new EventEmitter<any>();
  @Input() policy;
  // eslint-disable-next-line @typescript-eslint/naming-convention
  PolicyType = PolicyType;

  constructor(private analyticsService: AnalyticsService) {
    super();
  }

  onItemClick(coverage, index) {
    if (!this.isItemOpen(index)) {
      this.analyticsService.trackEvent(EventName.COVERAGE_SELECTED, Category.coverages, {
        name: coverage.shortName,
      });
    }
    this.toggleItemDetails(index);
  }
}
