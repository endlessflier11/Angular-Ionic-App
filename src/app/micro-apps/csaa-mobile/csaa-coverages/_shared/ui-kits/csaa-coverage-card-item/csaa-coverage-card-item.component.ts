import { Component, EventEmitter, Input, Output } from '@angular/core';
import { Agent } from '../../../../_core/interfaces/agent.interface';
import { Category, EventName } from '../../../../_core/interfaces/analytics.interface';
import { Coverage, PolicyType } from '../../../../_core/interfaces/policy.interface';
import { GlobalStateService, RouterService } from '../../../../_core/services';
import { AnalyticsService } from '../../../../_core/services/analytics.service';

export interface CoverageItemClickEvent {
  data: Coverage;
  event: any;
}

@Component({
  selector: 'csaa-coverage-card-item, csaa-indented-coverage-card-item',
  styleUrls: ['./csaa-coverage-card-item.component.scss'],
  templateUrl: 'csaa-coverage-card-item.component.html',
})
export class CsaaCoverageCardItemComponent {
  @Input() coverage: Coverage;
  @Input() policyNumber: string;
  @Input() policyType: PolicyType;
  @Input() agent: Agent;
  @Input() isOpen = false;
  @Output() itemClick = new EventEmitter<CoverageItemClickEvent>();

  // eslint-disable-next-line @typescript-eslint/naming-convention
  PolicyType = PolicyType;

  constructor(
    private readonly analyticsService: AnalyticsService,
    private readonly routerService: RouterService,
    private readonly globalStateService: GlobalStateService
  ) {}

  isComprehensiveCoverage(): boolean {
    return this.coverage.code === 'COMP';
  }

  isGlassIndicatorSet(): boolean {
    const ind = this.coverage.extensions.find((elem) => elem.attrName === 'glassInd');
    if (ind && ind.attrValue) {
      return true;
    }

    return false;
  }

  async goToIndentedCoverages(event) {
    event.stopPropagation();
    event.preventDefault();

    this.analyticsService.trackEvent(EventName.INDENTED_COVERAGES_VIEWED, Category.coverages, {
      policy_number: this.policyNumber,
    });

    this.globalStateService.setIndentedCoverageDetailsState({
      coverage: this.coverage,
      agent: this.agent,
      policyNumber: this.policyNumber,
      policyType: this.policyType,
    });

    this.routerService.navigateForward('csaa.coverages.indented', {
      policyNumber: this.policyNumber,
    });

    //   TODO V5: create CsaaIndentedCoverageDetailsPage
    //   this.modalController.create({
    //     component: CsaaIndentedCoverageDetailsPage,
    //   componentProps:
    //   });
    // }
  }

  handleItemClick(event) {
    this.itemClick.emit({ data: this.coverage, event });
  }
}
