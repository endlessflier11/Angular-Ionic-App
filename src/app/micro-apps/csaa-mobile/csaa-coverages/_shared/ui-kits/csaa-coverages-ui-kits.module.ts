import { NgModule } from '@angular/core';
import { CsaaPupCoveragesCardComponent } from './csaa-pup-coverages-card/csaa-pup-coverages-card.component';
import { CsaaGeneralCoveragesCardComponent } from './csaa-general-coverages-card/csaa-general-coverages-card.component';
import { CsaaDriversCoveragesCardComponent } from './csaa-drivers-coverages-card/csaa-drivers-coverages-card.component';
import { CsaaHomeCoveragesCardComponent } from './csaa-home-coverages-card/csaa-home-coverages-card.component';
import { CsaaVehiclesCoveragesCardComponent } from './csaa-vehicles-coverages-card/csaa-vehicles-coverages-card.component';
import { UiKitsModule } from '../../../_core/ui-kits/ui-kits.module';
import { IonicModule } from '@ionic/angular';
import { CommonModule } from '@angular/common';
import { CsaaCoverageCardItemComponent } from './csaa-coverage-card-item/csaa-coverage-card-item.component';
import {
  CsaaCoveragesVehicleCoveredCardComponent
} from './csaa-vehicle-coverages-covered-card/csaa-coverages-vehicle-covered-card.component';
import {
  CsaaCoveragesVehicleNotCoveredCardComponent
} from './csaa-vehicle-coverages-not-covered-card/csaa-coverages-vehicle-not-covered-card.component';
import { CsaaDriversCoveragesContentV1Component } from './csaa-drivers-coverages-content-v1/csaa-drivers-coverages-content-v1.component';
import { CsaaDriversCoveragesContentV2Component } from './csaa-drivers-coverages-content-v2/csaa-drivers-coverages-content-v2.component';

const SHARED_COMPONENTS = [
  CsaaCoverageCardItemComponent,
  CsaaPupCoveragesCardComponent,
  CsaaGeneralCoveragesCardComponent,
  CsaaDriversCoveragesCardComponent,
  CsaaDriversCoveragesContentV1Component,
  CsaaHomeCoveragesCardComponent,
  CsaaVehiclesCoveragesCardComponent,
  CsaaCoveragesVehicleCoveredCardComponent,
  CsaaCoveragesVehicleNotCoveredCardComponent,
  CsaaDriversCoveragesContentV2Component,
];

@NgModule({
  declarations: [...SHARED_COMPONENTS],
  imports: [CommonModule, IonicModule, UiKitsModule],
  exports: [...SHARED_COMPONENTS],
})
export class CsaaCoveragesUiKitsModule {}
