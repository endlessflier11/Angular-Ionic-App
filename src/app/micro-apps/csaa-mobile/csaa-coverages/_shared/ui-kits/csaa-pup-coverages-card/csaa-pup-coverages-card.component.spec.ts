import { ComponentFixture, TestBed } from '@angular/core/testing';
import { CsaaCoverageCardItemComponent } from '../csaa-coverage-card-item/csaa-coverage-card-item.component';
import { By, generatePolicy, PageTestingModule } from '@app/testing';
import { IonicModule } from '@ionic/angular';
import { UiKitsModule } from '../../../../_core/ui-kits/ui-kits.module';
import { CommonModule } from '@angular/common';
import { CsaaPupCoveragesCardComponent } from './csaa-pup-coverages-card.component';
import { PUP_COVERAGES_FIXTURE } from '../../../../../../../testing/fixtures/coverages/pup.coverages.fixture';
import { PolicyType } from '../../../../_core/interfaces';

describe('CsaaPupCoveragesCardComponent', () => {
  let component: CsaaPupCoveragesCardComponent;
  let fixture: ComponentFixture<CsaaPupCoveragesCardComponent>;
  const POLICY_NUMBER = 'CAPU123456789';

  beforeEach(async () => {
    try {
      await TestBed.configureTestingModule({
        declarations: [CsaaPupCoveragesCardComponent, CsaaCoverageCardItemComponent],

        imports: [
          CommonModule,
          IonicModule,
          UiKitsModule,
          PageTestingModule.withConfig({ providesStorage: true, providesAnalytics: true }),
        ],
      }).compileComponents();

      fixture = TestBed.createComponent(CsaaPupCoveragesCardComponent);
      component = fixture.componentInstance;
      const policy = generatePolicy({ number: POLICY_NUMBER, type: PolicyType.PUP });
      component.policy = policy;
      component.policyNumber = policy.number;
      component.coverages = PUP_COVERAGES_FIXTURE;

      fixture.detectChanges();
      await fixture.whenStable();
    } catch (error) {
      console.log(error);
    }
  });
  it('should create', () => {
    expect(component).toBeTruthy();
  });
  it('should match snapshots when there are coverages', async () => {
    fixture.detectChanges();
    await fixture.whenStable();
    expect(fixture).toMatchSnapshot();
  });
  it('should match snapshots when no coverages', () => {
    component.coverages = [];
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
  it('it should display one coverage item', () => {
    component.coverages = [component.coverages[0]];
    fixture.detectChanges();
    expect(fixture).toBeTruthy();
  });
  it('it should expand coverage card item ', () => {
    const coverageItem = fixture.debugElement.query(By.css('.csaa-coverage-card-item'));
    expect(coverageItem.query(By.css('.unopened-card'))).toBeTruthy();
    expect(coverageItem.query(By.css('.description'))).toBeFalsy();
    coverageItem.triggerEventHandler('click', null);
    fixture.detectChanges();
    expect(coverageItem.query(By.css('.unopened-card'))).toBeFalsy();
    expect(coverageItem.query(By.css('.description'))).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });
  it('it should collapse the card after clicking on the expanded card', () => {
    const coverageItem = fixture.debugElement.query(By.css('.csaa-coverage-card-item'));
    expect(coverageItem.query(By.css('.unopened-card'))).toBeTruthy();
    expect(coverageItem.query(By.css('.description'))).toBeFalsy();
    coverageItem.triggerEventHandler('click', null);
    fixture.detectChanges();
    expect(coverageItem.query(By.css('.unopened-card'))).toBeFalsy();
    expect(coverageItem.query(By.css('.description'))).toBeTruthy();
    coverageItem.triggerEventHandler('click', null);
    fixture.detectChanges();
    expect(coverageItem.query(By.css('.unopened-card'))).toBeTruthy();
    expect(coverageItem.query(By.css('.description'))).toBeFalsy();
    expect(fixture).toMatchSnapshot();
  });
});
