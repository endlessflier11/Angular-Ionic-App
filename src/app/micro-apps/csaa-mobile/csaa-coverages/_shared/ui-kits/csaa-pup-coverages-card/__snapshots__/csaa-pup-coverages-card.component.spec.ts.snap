// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CsaaPupCoveragesCardComponent it should collapse the card after clicking on the expanded card 1`] = `
<csaa-pup-coverages-card
  PolicyType={[Function Object]}
  __ngContext__={[Function LRootView]}
  analyticsService={[Function AnalyticsMockService]}
  coverages={[Function Array]}
  loading="false"
  openCard="undefined"
  policy={[Function Object]}
  policyNumber={[Function String]}
  viewDeclarationClick={[Function EventEmitter_]}
>
  <ion-card
    class="coverage-list"
  >
    <ion-card-header>
      <ion-card-title>
        <div
          class="display-flex ion-align-items-center"
        >
           CAPU123456789 
        </div>
        <csaa-theme-icon>
          <img
            src="/assets/csaa-mobile/vectors/icn-PUP.svg"
            style="width: 40px; height: 40px;"
          />
        </csaa-theme-icon>
      </ion-card-title>
      <ion-card-subtitle>
        Personal Umbrella Policy
      </ion-card-subtitle>
    </ion-card-header>
    <ion-card-content
      class="csaa-item-list"
    >
      <ion-list>
        <ion-item>
          <csaa-coverage-card-item>
            <div
              class="csaa-coverage-card-item"
            >
              <h4
                class="title"
              >
                Personal Umbrella
              </h4>
              <div
                class="unopened-card"
              >
                <p
                  class="limit-info"
                >
                  <span>
                    $1,000,000
                  </span>
                </p>
              </div>
            </div>
          </csaa-coverage-card-item>
        </ion-item>
      </ion-list>
      <ion-button
        expand="full"
      >
         View Declarations
      </ion-button>
    </ion-card-content>
  </ion-card>
</csaa-pup-coverages-card>
`;

exports[`CsaaPupCoveragesCardComponent it should expand coverage card item  1`] = `
<csaa-pup-coverages-card
  PolicyType={[Function Object]}
  __ngContext__={[Function LRootView]}
  analyticsService={[Function AnalyticsMockService]}
  coverages={[Function Array]}
  loading="false"
  openCard="0"
  policy={[Function Object]}
  policyNumber={[Function String]}
  viewDeclarationClick={[Function EventEmitter_]}
>
  <ion-card
    class="coverage-list"
  >
    <ion-card-header>
      <ion-card-title>
        <div
          class="display-flex ion-align-items-center"
        >
           CAPU123456789 
        </div>
        <csaa-theme-icon>
          <img
            src="/assets/csaa-mobile/vectors/icn-PUP.svg"
            style="width: 40px; height: 40px;"
          />
        </csaa-theme-icon>
      </ion-card-title>
      <ion-card-subtitle>
        Personal Umbrella Policy
      </ion-card-subtitle>
    </ion-card-header>
    <ion-card-content
      class="csaa-item-list"
    >
      <ion-list>
        <ion-item>
          <csaa-coverage-card-item>
            <div
              class="csaa-coverage-card-item open"
            >
              <h4
                class="title"
              >
                Personal Umbrella
              </h4>
              <div
                class="description"
              >
                Excess Liability Coverage on Auto and Home Policies
              </div>
            </div>
          </csaa-coverage-card-item>
        </ion-item>
      </ion-list>
      <ion-button
        expand="full"
      >
         View Declarations
      </ion-button>
    </ion-card-content>
  </ion-card>
</csaa-pup-coverages-card>
`;

exports[`CsaaPupCoveragesCardComponent should match snapshots when no coverages 1`] = `
<csaa-pup-coverages-card
  PolicyType={[Function Object]}
  __ngContext__={[Function LRootView]}
  analyticsService={[Function AnalyticsMockService]}
  coverages={[Function Array]}
  loading="false"
  openCard="undefined"
  policy={[Function Object]}
  policyNumber={[Function String]}
  viewDeclarationClick={[Function EventEmitter_]}
>
</csaa-pup-coverages-card>
`;

exports[`CsaaPupCoveragesCardComponent should match snapshots when there are coverages 1`] = `
<csaa-pup-coverages-card
  PolicyType={[Function Object]}
  __ngContext__={[Function LRootView]}
  analyticsService={[Function AnalyticsMockService]}
  coverages={[Function Array]}
  loading="false"
  openCard="undefined"
  policy={[Function Object]}
  policyNumber={[Function String]}
  viewDeclarationClick={[Function EventEmitter_]}
>
  <ion-card
    class="coverage-list"
  >
    <ion-card-header>
      <ion-card-title>
        <div
          class="display-flex ion-align-items-center"
        >
           CAPU123456789 
        </div>
        <csaa-theme-icon>
          <img
            src="/assets/csaa-mobile/vectors/icn-PUP.svg"
            style="width: 40px; height: 40px;"
          />
        </csaa-theme-icon>
      </ion-card-title>
      <ion-card-subtitle>
        Personal Umbrella Policy
      </ion-card-subtitle>
    </ion-card-header>
    <ion-card-content
      class="csaa-item-list"
    >
      <ion-list>
        <ion-item>
          <csaa-coverage-card-item>
            <div
              class="csaa-coverage-card-item"
            >
              <h4
                class="title"
              >
                Personal Umbrella
              </h4>
              <div
                class="unopened-card"
              >
                <p
                  class="limit-info"
                >
                  <span>
                    $1,000,000
                  </span>
                </p>
              </div>
            </div>
          </csaa-coverage-card-item>
        </ion-item>
      </ion-list>
      <ion-button
        expand="full"
      >
         View Declarations
      </ion-button>
    </ion-card-content>
  </ion-card>
</csaa-pup-coverages-card>
`;
